richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichTextState.kt

```diff
@@ -304,20 +304,37 @@ public class RichTextState internal constructor(
         textRange: TextRange,
         text: String
     ) {
-        require(textRange.min >= 0) {
-            "The start index must be non-negative."
-        }
+        // Ensure that the start and end indices are within bounds
+        val start = textRange.min.coerceAtLeast(0) // Ensure the start is non-negative
+        val end = textRange.max.coerceAtMost(textFieldValue.text.length) // Ensure the end does not exceed text length
 
-        require(textRange.max <= textFieldValue.text.length) {
-            "The end index must be within the text bounds. " +
-                    "The text length is ${textFieldValue.text.length}, " +
-                    "but the end index is ${textRange.max}."
+        require(start <= end) {
+            "The start index ($start) must be less than or equal to the end index ($end)."
         }
 
-        removeTextRange(textRange)
-        addTextAfterSelection(text = text)
+        // Perform the replacement with safe slicing
+        val beforeText = textFieldValue.text.safeSubstring(0, start)
+        val afterText = textFieldValue.text.safeSubstring(end)
+        val newText = beforeText + text + afterText
+
+        onTextFieldValueChange(
+            newTextFieldValue = textFieldValue.copy(
+                text = newText,
+                selection = TextRange(start + text.length), // Set cursor to the end of the inserted text
+            )
+        )
+    }
+
+    // A helper function to safely slice strings without throwing exceptions
+    private fun String.safeSubstring(startIndex: Int, endIndex: Int = this.length): String {
```

Comment:
This helper function is useful, you can keep it even if it's going to be unused.